File: ProductCatalogFrontend/src/app/services/serviceability.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';

@Injectable({
  providedIn: 'root',
})
export class ServiceabilityService {
  serviceability: any;
  baseURL = 'http://localhost:8082/service';
  constructor(private http: HttpClient) {}

  addServiceability(formData: FormData) {
    return this.http.post(`${this.baseURL}`, formData);
  }
  getServiceability(pincode, productcode) {
    return this.http.get(`${this.baseURL}/${pincode}/${productcode}`);
  }
}

File: ProductCatalogBackend/src/main/java/com/nagarro/service/ServiceabilityService.java
package com.nagarro.service;

import com.nagarro.model.Serviceability;

public interface ServiceabilityService {
	/**
	 * Find serviceability corresponds to producutcode and pincode
	 * 
	 * @param productCode
	 * @param pincode
	 * @return
	 */
	public Serviceability findServiceabilityByProductCodeAndPincode(String productCode, Integer pincode);

	/**
	 * Add serviceability
	 * 
	 * @param productCode
	 * @param pincode
	 * @param expectedDeliveryTime
	 */
	public void addServiceability(String productCode, Integer pincode, Integer expectedDeliveryTime);
}
